### ISSUE TITLE
[RCV] Constant/Common RCV Rollbacks

### ISSUE THREAD

**Issue by PhantomWolf42** [MrBlizzard/RCAdmins-Tracker#13]
RC Vaults don't appear to save on open/close and are subject to rollback without warning.

**Comment by MrBlizzard** [comment]
Need more info, next case let me know more please

**Comment by PhantomWolf42** [comment]
Recent occurrance: http://www.rc-gamers.com/admin/editmodule/index/editoraction/index/preset/27511969?ticket_id=a8a26bf3

http://www.rc-gamers.com/forum/m/7925921/viewthread/31853154-rcv-page-1-wiped

Appears an entire RCV page cleared of the contents. The player's other rcv pages, enderchest, inventory and placed chests, claims and trusted friends were also checked. There were no signs of the items elsewhere.

**Comment by PhantomWolf42** [comment]
http://www.rc-gamers.com/admin/editmodule/index/editoraction/index/preset/27511969?ticket_id=c473fbcd

Sometimes it appears that a small rollback occurs, deleting only the most recently added items. In other circumstances full rcv pages disappear.

**Comment by LilyMu122** [comment]
A player on Pixel 3 mentioned that they believe shift-clicking items into their RCV quickly may be causing issues.

**Comment by MrBlizzard** [comment]
There seem to be a few separate bugs, one of which has been patched. 

1. The patched bug was able to be recreated easily. It can be clearly seen in this video https://www.youtube.com/watch?v=5IIPpOLSinE&feature=youtu.be .  Tl;dr, after 1 min restart message, keep vault open and put things in inven. Keep it open until kick, log back in, and the items would be back in the rcv and in the inventory.

2. Another bug appears with shift clicking, which can bug out the items in rcv and some can appear to be lost. I was able to recreate the buggy nature by filling a page with 1 quick ball and spamming nonstop, until finally some started lagging behind. I didn't lose items though. Upon the patch that was intended to fix the other issue above ^, I wasn't able to recreate the buggy nature. 
**This could use testing on every pixel to see if they can recreate it. Here is a video depicting what happened** https://www.youtube.com/watch?v=VC-JozNQvcw&feature=youtu.be

3. This is a little vague, not a lot of info, but after the patch, several people reported losing items upon putting them in rcv and it rolling back. Test patch uploaded to p5 which has removed all code that could potentially cause items being lost. It's a big change, so it's on p5 for the next day to see if any unintended side effects become present.

Tl;dr, Point 1 should be fixed, Point 2 could use testing to see if it still occurs, and Point 3 is being worked on, but more info would be nice to recreate it.

**Comment by MrBlizzard** [comment]
New Patch was put on p5 on monday, and it seems to have fixed all problems. No oddities have become known and no one on p5 has reported issues.

**Comment by MrBlizzard** [comment]
Should be fixed and uploaded to all
